### PR TITLE
Zshen/feature/tdq 12397 sort data on result of column set final

### PR DESCRIPTION
**Pull request by zshen-talend**

### CODE REVIEW

File: main/plugins/org.talend.dataprofiler.common.ui/src/org/talend/dataprofiler/common/ui/pagination/controller/PageableWithIndexController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2016 Talend Inc. - www.talend.com
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataprofiler.common.ui/src/org/talend/dataprofiler/common/ui/pagination/controller/PageableWithIndexController.java]: should be 2006-2018, need to make sure the java template is up to date

File: main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/ColumnSetDBMap.java [hunk 2]
```diff
@@ -103,30 +108,49 @@ public ColumnSetDBMap() {
...
+        int reorderIndex = ((DataValidationImpl) dataValiator).getReorderIndex();
+        if (reorderIndex > -1) {
+            int sortDirection = ((DataValidationImpl) dataValiator).getSortDirection();
+            Comparator<?> dbMapSpecialColCompartor =
+                    new DBMapSpecialColCompartor(((DataValidationImpl) dataValiator).getReorderIndex());
+            if (sortDirection == 0) {
+                dbMapSpecialColCompartor = Collections.reverseOrder(dbMapSpecialColCompartor);
+            }
+            iterator =
+                    Pump.sort(this.keySet().iterator(), false, 2000, dbMapSpecialColCompartor,
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/ColumnSetDBMap.java]: what's the meaning of 2000?

File: main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/compartor/DBMapSpecialColCompartor.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// ============================================================================
+//
+// Copyright (C) 2006-2016 Talend Inc. - www.talend.com
```
Comment (xqliu-talend) [inline main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/compartor/DBMapSpecialColCompartor.java]: should be 2006-2018, need to make sure the java template is up to date

File: main/plugins/org.talend.dataprofiler.common.ui/src/org/talend/dataprofiler/common/ui/pagination/controller/PageableWithIndexController.java (hunk 1 above)
Comment (zshen-talend) [reply main/plugins/org.talend.dataprofiler.common.ui/src/org/talend/dataprofiler/common/ui/pagination/controller/PageableWithIndexController.java]: modify it

File: main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/ColumnSetDBMap.java (hunk 2 above)
Comment (zshen-talend) [reply main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/ColumnSetDBMap.java]: add a new variable and add a commont

File: main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/compartor/DBMapSpecialColCompartor.java (hunk 3 above)
Comment (zshen-talend) [reply main/plugins/org.talend.dataquality/src/org/talend/dataquality/indicators/mapdb/compartor/DBMapSpecialColCompartor.java]: modify it

**Pull request merged**